promptoglyph-path.d

```diff
@@ -38,15 +46,51 @@ void main(string[] args)
 		writeAndFail(ex.msg, "\n", helpString);
 	}
 
-
-	immutable string home = environment["HOME"].ifThrown("");
-	immutable string cwd = getcwd().ifThrown(environment["PWD"]).ifThrown("???");
+	// NOTE(dkg): on Windows (under Cygwin even) env["HOME"] will include
+	//            the drive letter and the path is a Windows style path
+	// Another problem is that home and cwd will result in
+	// home, cwd: C:\Cygwin\home\dkg, C:\Users\dkg\Projekte\d\promptd
+	// in Cygwin, so the homeToTilde function will not work here.
+	// When you convert both paths via cygpath to Unix style path they are 
+	// different as well:
+	// home, cwd: /home/dkg, /cygdrive/c/Users/dkg/Projekte/d/promptd
+	version(PathFixNeeded)
```

Comment:
Can we hoist this into `systempath.d` so we don't have to sprinkle `version (PathFixNeeded)` around this file as well?